kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -254,4 +270,25 @@ private void validate(String value) {
     private static void addConfig(HashMap<String, TableConfig<?>> configs, TableConfig<?> config) {
         configs.put(config.getKey().toLowerCase(Locale.ROOT), config);
     }
+
+    private static Map<String, String> parseJSONKeyValueMap(Engine engine, String jsonString) {
```

Comment:
Please add a docstring and explain the limitations of this method --- things like how this will only work with "string" -> "string" maps.